packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -457,4 +474,268 @@ linter:
+        String event = response['event'];
+        dynamic params = response['params'];
+
+        if (params is Map) {
+          if (event == 'server.status')
+            _handleStatus(response['params']);
+          else if (event == 'analysis.errors')
+            _handleAnalysisIssues(response['params']);
+          else if (event == 'server.error')
+            _handleServerError(response['params']);
+        }
+      } else if (response['error'] != null) {
+        printError('Error from the analysis server: ${response['error']['message']}');
+      }
+    }
+  }
+
+  void _handleStatus(Map<String, dynamic> statusInfo) {
+    // {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
+    if (statusInfo['analysis'] != null) {
+      bool isAnalyzing = statusInfo['analysis']['isAnalyzing'];
+      _analyzingController.add(isAnalyzing);
+    }
+  }
+
+  void _handleServerError(Map<String, dynamic> errorInfo) {
+    printError('Error from the analysis server: ${errorInfo['message']}');
+  }
+
+  void _handleAnalysisIssues(Map<String, dynamic> issueInfo) {
+    // {"event":"analysis.errors","params":{"file":"/Users/.../lib/main.dart","errors":[]}}
+    String file = issueInfo['file'];
+    List<AnalysisError> errors = issueInfo['errors'].map((Map<String, dynamic> json) => new AnalysisError(json)).toList();
+    _errorsController.add(new FileAnalysisErrors(file, errors));
+  }
+
+  Future dispose() async => _process?.kill();
+}
+
+class FileAnalysisErrors {
+  FileAnalysisErrors(this.file, this.errors);
+
+  final String file;
+  final List<AnalysisError> errors;
+}
+
+class AnalysisError implements Comparable<AnalysisError> {
+  AnalysisError(this.json);
+
+  static final Map<String, int> _severityMap = <String, int> {
+    'ERROR': 3,
+    'WARNING': 2,
+    'INFO': 1
+  };
+
+  // "severity":"INFO","type":"TODO","location":{
+  //   "file":"/Users/.../lib/test.dart","offset":362,"length":72,"startLine":15,"startColumn":4
+  // },"message":"...","hasFix":false}
+  Map<String, dynamic> json;
+
+  String get severity => json['severity'];
+  int get severityLevel => _severityMap[severity] ?? 0;
+  String get type => json['type'];
+  String get message => json['message'];
+
+  String get file => json['location']['file'];
+  int get startLine => json['location']['startLine'];
+  int get startColumn => json['location']['startColumn'];
+
+  int compareTo(AnalysisError other) {
+    // Sort in order of severity, file path, source line, and message.
+    int diff = other.severityLevel - severityLevel;
+    if (diff != 0)
+      return diff;
+    if (file != other.file)
+      return file.compareTo(other.file);
+    if (startLine != other.startLine)
+      return startLine - other.startLine;
+    return message.compareTo(other.message);
+  }
+
+  String toString() {
+    String relativePath = path.relative(file);
+    return '${severity.toLowerCase().padLeft(7)} • $message • $relativePath:$startLine:$startColumn';
```

Comment:
We should try to match the output of the analyzer in single-file mode (either by changing that output, or changing this output, but they should be consistent)